### PR TITLE
implemented KeyboardInterrupt try-except in main loop

### PR DESCRIPTION
**Pull request by honeykjoule**
addresses #12 by allowing user to exit loop with keyboard interrupt command. the interrupt is implemented in a try-except block on the main loop.

there are other ideas to address the indefinite loop problem, such as setting a max iteration param that halts when the counter reaches a certain level. 

this fix is a simple, immediate remedy to help users feel confident they haven't left their script running to burn through their credit budget.

meow.

### CODE REVIEW

**Review by honeykjoule — COMMENTED** [review]
fixed indentation in the while loop in the try-except. now sleep, gn.

File: babyagi.py [hunk 1]
```diff
@@ -102,37 +102,40 @@ def context_agent(query: str, index: str, n: int):
...
+
+        for new_task in new_tasks:
+            task_id_counter += 1
+            new_task.update({"task_id": task_id_counter})
+            add_task(new_task)
+        prioritization_agent(this_task_id)
+except KeyboardInterrupt:
+    print("\nLoop interrupted by user. Exiting.")
 
 time.sleep(1)  # Sleep before checking the task list again
```
Comment (QuentMe) [inline babyagi.py:141]: sleep is outside of the main loop

**Comment by francip** [comment]
I'd like to pick this one, but can you first resolve the conflicts?

File: babyagi.py (hunk 1 above)
Comment (honeykjoule) [reply babyagi.py]: fixed

**Comment by honeykjoule** [comment]
@francip thanks, resolved conflicts and moved sleep into loop (@QuentMe )

**Comment by honeykjoule** [comment]
closing this because it my commit looks bugged. will reopen with a clean commit

**Pull request closed without merging**